src/content-script.ts

```diff
@@ -1,5 +1,23 @@
-function injectScript(url: string) {
+import { getSettings } from "./settings";
+
+declare module globalThis {
+	function cloneInto(data: any, view: any): any;
+}
+
+// Firefox requires cloneInto() to send data to injected script,
+// Chrome does not provide this function and automatically clones
+// the data. To not trigger a ReferenceError in Chrome, we polyfill it here.
+globalThis.cloneInto ??= data => data;
+
+async function injectScript(url: string) {
+	const settings = await getSettings();
 	const script = document.createElement("script");
+	script.onload = function() {
+		document.dispatchEvent(new CustomEvent(
+			"github-monaco-get-settings",
+			{ detail: globalThis.cloneInto(settings, document.defaultView) },
+		));
+	};
```

Comment:
What do you think of json encoding the settings in `document.head.dataset.hedietMonacoEditorSettings`?
(like `hedietMonacoEditorPublicPath`)

This would simplify the loading mechanism & get rid of the `cloneInto` problematic.

Reply:
`dataset` can only carry strings, unfortunately.

Reply:
Yes, but you can Json serialize your data to make any object a string and later json deserialize it!
But I'm open to the event approach if it is the recommended/default way. How did you came up with that idea?

Reply:
> Yes, but you can Json serialize your data to make any object a string and later json deserialize it!

I don't know how I didn't realized that! 😲 Will update the PR.

> How did you came up with that idea?

Some SO answer, can't find now.